### ISSUE TITLE
Duplicati.GUI.TrayIcon has stopped working after upgrade, Duplicati.Server crashes

### ISSUE THREAD

**Issue by travisdietrich** [duplicati/duplicati#2608]
I have:
- [x] searched open and closed issues for duplicates

----------------------------------------

### Version info

**Duplicati Version:** 2.0.1.57 - 2.0.1.62
**Operating System:** Windows Server 2012 R2 x64
**Backend:**

### Bug description
After updating, server will not start and tray icon crashes.  I can downgrade back to 2.0.1.56 to resolve the issue, but this is the last version that will work.

### Steps to reproduce
- Start Duplicati.GUI.TrayIcon.exe

Also tried:
- Start Duplicati.Server.exe

### Screenshots
Error Message:
![image](https://user-images.githubusercontent.com/16649815/28492422-22aae766-6ec9-11e7-909c-20ef50e25fbc.png)

Error Message Text:
```
Problem signature:
  Problem Event Name:	CLR20r3
  Problem Signature 01:	Duplicati.GUI.TrayIcon.exe
  Problem Signature 02:	2.0.1.62
  Problem Signature 03:	592fdd7b
  Problem Signature 04:	Duplicati.GUI.TrayIcon
  Problem Signature 05:	2.0.1.72
  Problem Signature 06:	595ca1e9
  Problem Signature 07:	2c
  Problem Signature 08:	7
  Problem Signature 09:	System.TypeInitialization
  OS Version:	6.3.9600.2.0.0.400.8
  Locale ID:	1033
  Additional Information 1:	2b4f
  Additional Information 2:	2b4f62bd8021344d8a58f9747f1c1408
  Additional Information 3:	89ac
  Additional Information 4:	89ac39d0098024f245cbeffc3b68ca3f
```

### debug log duplicati.gui.trayicon
Event Log 1 Duplicati.GUI.TrayIcon:
```
Log Name:      Application
Source:        .NET Runtime
Date:          7/22/2017 10:29:09 AM
Event ID:      1026
Task Category: None
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      TheJibbers
Description:
Application: Duplicati.GUI.TrayIcon.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.IO.FileLoadException

Exception Info: System.IO.FileLoadException
   at Duplicati.Library.AutoUpdater.UpdaterManager..cctor()

Exception Info: System.TypeInitializationException
   at Duplicati.Library.AutoUpdater.UpdaterManager.set_RequiresRespawn(Boolean value)
   at Duplicati.GUI.TrayIcon.Program.Main(String[] args)
   at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
   at System.AppDomain.ExecuteAssemblyByName(String assemblyName, String[] args)
   at System.AppDomain.ExecuteAssemblyByName(System.String, System.String[])
   at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent(System.Reflection.MethodInfo, System.String[], Duplicati.Library.AutoUpdater.AutoUpdateStrategy)
   at Duplicati.GUI.TrayIcon.Program.Main(System.String[])


Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name=".NET Runtime" />
    <EventID Qualifiers="0">1026</EventID>
    <Level>2</Level>
    <Task>0</Task>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2017-07-22T15:29:09.000000000Z" />
    <EventRecordID>39417</EventRecordID>
    <Channel>Application</Channel>
    <Computer>TheJibbers</Computer>
    <Security />
  </System>
  <EventData>
    <Data>Application: Duplicati.GUI.TrayIcon.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.IO.FileLoadException

Exception Info: System.IO.FileLoadException
   at Duplicati.Library.AutoUpdater.UpdaterManager..cctor()

Exception Info: System.TypeInitializationException
   at Duplicati.Library.AutoUpdater.UpdaterManager.set_RequiresRespawn(Boolean value)
   at Duplicati.GUI.TrayIcon.Program.Main(String[] args)
   at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
   at System.AppDomain.ExecuteAssemblyByName(String assemblyName, String[] args)
   at System.AppDomain.ExecuteAssemblyByName(System.String, System.String[])
   at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent(System.Reflection.MethodInfo, System.String[], Duplicati.Library.AutoUpdater.AutoUpdateStrategy)
   at Duplicati.GUI.TrayIcon.Program.Main(System.String[])

</Data>
  </EventData>
</Event>
```

Event Log 2 Duplicati.GUI.TrayIcon:
```
Log Name:      Application
Source:        Application Error
Date:          7/22/2017 10:29:09 AM
Event ID:      1000
Task Category: (100)
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      TheJibbers
Description:
Faulting application name: Duplicati.GUI.TrayIcon.exe, version: 2.0.1.62, time stamp: 0x592fdd7b
Faulting module name: KERNELBASE.dll, version: 6.3.9600.18666, time stamp: 0x58f33794
Exception code: 0xe0434352
Fault offset: 0x00000000000095fc
Faulting process id: 0x185c
Faulting application start time: 0x01d302ff41e7d10e
Faulting application path: C:\Program Files\Duplicati 2\Duplicati.GUI.TrayIcon.exe
Faulting module path: C:\Windows\system32\KERNELBASE.dll
Report Id: 80e488c0-6ef2-11e7-8124-000423c11e97
Faulting package full name: 
Faulting package-relative application ID: 
Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error" />
    <EventID Qualifiers="0">1000</EventID>
    <Level>2</Level>
    <Task>100</Task>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2017-07-22T15:29:09.000000000Z" />
    <EventRecordID>39418</EventRecordID>
    <Channel>Application</Channel>
    <Computer>TheJibbers</Computer>
    <Security />
  </System>
  <EventData>
    <Data>Duplicati.GUI.TrayIcon.exe</Data>
    <Data>2.0.1.62</Data>
    <Data>592fdd7b</Data>
    <Data>KERNELBASE.dll</Data>
    <Data>6.3.9600.18666</Data>
    <Data>58f33794</Data>
    <Data>e0434352</Data>
    <Data>00000000000095fc</Data>
    <Data>185c</Data>
    <Data>01d302ff41e7d10e</Data>
    <Data>C:\Program Files\Duplicati 2\Duplicati.GUI.TrayIcon.exe</Data>
    <Data>C:\Windows\system32\KERNELBASE.dll</Data>
    <Data>80e488c0-6ef2-11e7-8124-000423c11e97</Data>
    <Data>
    </Data>
    <Data>
    </Data>
  </EventData>
</Event>
```

### debug log duplicati.server

Command Line Duplicati.Server:
```
C:\Program Files\Duplicati 2>Duplicati.Server.exe

Unhandled Exception: System.TypeInitializationException: The type initializer fo
r 'Duplicati.Library.AutoUpdater.UpdaterManager' threw an exception. ---> System
.IO.FileLoadException: Could not load file or assembly 'Duplicati.Library.Utilit
y, Version=2.0.1.62, Culture=neutral, PublicKeyToken=8bfe994a39631a7b' or one of
 its dependencies. The located assembly's manifest definition does not match the
 assembly reference. (Exception from HRESULT: 0x80131040) ---> System.IO.FileLoa
dException: Could not load file or assembly 'Duplicati.Library.Utility, Version=
2.0.1.72, Culture=neutral, PublicKeyToken=8bfe994a39631a7b' or one of its depend
encies. The located assembly's manifest definition does not match the assembly r
eference. (Exception from HRESULT: 0x80131040)

   --- End of inner exception stack trace ---
   at Duplicati.Library.AutoUpdater.UpdaterManager..cctor()
   --- End of inner exception stack trace ---
   at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent(MethodInfo
method, String[] cmdargs, AutoUpdateStrategy defaultstrategy)
   at Duplicati.Server.Program.Main(String[] args)
   at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args
)
   at System.AppDomain.ExecuteAssemblyByName(String assemblyName, String[] args)

   at System.AppDomain.ExecuteAssemblyByName(String assemblyName, String[] args)

   at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent(MethodInfo
method, String[] cmdargs, AutoUpdateStrategy defaultstrategy)
   at Duplicati.Server.Program.Main(String[] args)

C:\Program Files\Duplicati 2>
```

Event Log 3 Duplicati.Server:
```
Log Name:      Application
Source:        .NET Runtime
Date:          7/22/2017 10:38:15 AM
Event ID:      1026
Task Category: None
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      TheJibbers
Description:
Application: Duplicati.Server.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.IO.FileLoadException

Exception Info: System.IO.FileLoadException
   at Duplicati.Library.AutoUpdater.UpdaterManager..cctor()

Exception Info: System.TypeInitializationException
   at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent(MethodInfo method, String[] cmdargs, AutoUpdateStrategy defaultstrategy)
   at Duplicati.Server.Program.Main(String[] args)
   at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
   at System.AppDomain.ExecuteAssemblyByName(String assemblyName, String[] args)
   at System.AppDomain.ExecuteAssemblyByName(System.String, System.String[])
   at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent(System.Reflection.MethodInfo, System.String[], Duplicati.Library.AutoUpdater.AutoUpdateStrategy)
   at Duplicati.Server.Program.Main(System.String[])


Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name=".NET Runtime" />
    <EventID Qualifiers="0">1026</EventID>
    <Level>2</Level>
    <Task>0</Task>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2017-07-22T15:38:15.000000000Z" />
    <EventRecordID>39423</EventRecordID>
    <Channel>Application</Channel>
    <Computer>TheJibbers</Computer>
    <Security />
  </System>
  <EventData>
    <Data>Application: Duplicati.Server.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.IO.FileLoadException

Exception Info: System.IO.FileLoadException
   at Duplicati.Library.AutoUpdater.UpdaterManager..cctor()

Exception Info: System.TypeInitializationException
   at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent(MethodInfo method, String[] cmdargs, AutoUpdateStrategy defaultstrategy)
   at Duplicati.Server.Program.Main(String[] args)
   at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
   at System.AppDomain.ExecuteAssemblyByName(String assemblyName, String[] args)
   at System.AppDomain.ExecuteAssemblyByName(System.String, System.String[])
   at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent(System.Reflection.MethodInfo, System.String[], Duplicati.Library.AutoUpdater.AutoUpdateStrategy)
   at Duplicati.Server.Program.Main(System.String[])

</Data>
  </EventData>
</Event>
```

Event Log 4 Duplicati.Server:
```
Log Name:      Application
Source:        Application Error
Date:          7/22/2017 10:38:15 AM
Event ID:      1000
Task Category: (100)
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      TheJibbers
Description:
Faulting application name: Duplicati.Server.exe, version: 2.0.1.62, time stamp: 0x592fdd78
Faulting module name: KERNELBASE.dll, version: 6.3.9600.18666, time stamp: 0x58f33794
Exception code: 0xe0434352
Fault offset: 0x00000000000095fc
Faulting process id: 0x1f80
Faulting application start time: 0x01d3030087b98190
Faulting application path: C:\Program Files\Duplicati 2\Duplicati.Server.exe
Faulting module path: C:\Windows\system32\KERNELBASE.dll
Report Id: c6aa4d8f-6ef3-11e7-8124-000423c11e97
Faulting package full name: 
Faulting package-relative application ID: 
Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error" />
    <EventID Qualifiers="0">1000</EventID>
    <Level>2</Level>
    <Task>100</Task>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2017-07-22T15:38:15.000000000Z" />
    <EventRecordID>39424</EventRecordID>
    <Channel>Application</Channel>
    <Computer>TheJibbers</Computer>
    <Security />
  </System>
  <EventData>
    <Data>Duplicati.Server.exe</Data>
    <Data>2.0.1.62</Data>
    <Data>592fdd78</Data>
    <Data>KERNELBASE.dll</Data>
    <Data>6.3.9600.18666</Data>
    <Data>58f33794</Data>
    <Data>e0434352</Data>
    <Data>00000000000095fc</Data>
    <Data>1f80</Data>
    <Data>01d3030087b98190</Data>
    <Data>C:\Program Files\Duplicati 2\Duplicati.Server.exe</Data>
    <Data>C:\Windows\system32\KERNELBASE.dll</Data>
    <Data>c6aa4d8f-6ef3-11e7-8124-000423c11e97</Data>
    <Data>
    </Data>
    <Data>
    </Data>
  </EventData>
</Event>
```

**Comment by Dobatymo** [comment]
I had the same problem. Downloading the installer manually (and not using the build in update) and reinstalling help in my case.

**Comment by travisdietrich** [comment]
I tried all the installers from 2.0.1.56 forward and was not able to resolve.  Tried both the upgrade installation, and then the repair installation afterwards.

**Comment by KommX** [comment]
On one of my hosts (Windows 10) I think I have the same problem after updating to 2.0.1.73.
After starting Duplicati I can see "Duplicati.GUI.TrayIcon" in the task manager for some seconds. Then it disappears. I actually never see the tray icon.

**Comment by DracoriusRhonix** [comment]
For the record: I had the same problem as KommX. After updating to 2.0.1.73 through the web gui, Duplicati.GUI.TrayIcon failed to start. Reinstalling from the msi installer over the current install directory (which preserves settings aparently, phew) fixed the issue. 
This is on windows 10 x64 by the way.

**Comment by kenkendk** [comment]
@travisdietrich Does this also occur with the 2.0.1.73 version?
Can you try to see where the loading problem occurs:
http://www.hanselman.com/blog/BackToBasicsUsingFusionLogViewerToDebugObscureLoaderErrors.aspx

**Comment by travisdietrich** [comment]
I installed 2.0.1.73 and it does not have the error.  Sorry I was not looking at the experimental builds, looks like the issue was fixed (or went away).  Let me know if you want me to test further.

**Comment by kenkendk** [comment]
I think it is fixed. It was probably caused by a weird Assembly that was updated.

**Comment by sunwayru** [comment]
I had the same problem. Downloading the installer manually (and not using the build in update) and reinstalling NOT helped in my case: can't open localhost:8200

**Comment by mentalinc** [comment]
Also have this issue.
Installed version duplicati-2.0.2.1_beta_2017-08-01-x64.msi
Had issues with S3 Compatibility (issue  #2350), saw they were fixed in duplicati-2.0.2.3_canary_2017-09-08-x64.msi.

Changed the release to experimental, and tried to upgrade within the app. it just froze and got stuck trying to connect.

Tried to manually upgrade with the downloaded msi.
Same issue, cant start tray app.

Have uninstalled, uninstalled with Revo Uninstaller, rebooted between uninstalls/installs nothing works ends up with the same error messages below.

Ran 

>sfc /scannow

Beginning system scan.  This process will take some time.

Beginning verification phase of system scan.
Verification 100% complete.

Windows Resource Protection did not find any integrity violations.


Two errors in the event log:
.net runtime ID 1026

Application: Duplicati.GUI.TrayIcon.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.IO.FileLoadException

Exception Info: System.IO.FileLoadException
   at Duplicati.Library.AutoUpdater.UpdaterManager..cctor()

Exception Info: System.TypeInitializationException
   at Duplicati.Library.AutoUpdater.UpdaterManager.set_RequiresRespawn(Boolean value)
   at Duplicati.GUI.TrayIcon.Program.Main(String[] args)
   at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
   at System.AppDomain.ExecuteAssemblyByName(String assemblyName, String[] args)
   at System.AppDomain.ExecuteAssemblyByName(System.String, System.String[])
   at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent(System.Reflection.MethodInfo, System.String[], Duplicati.Library.AutoUpdater.AutoUpdateStrategy)
   at Duplicati.GUI.TrayIcon.Program.Main(System.String[])


Application Error 1000
Faulting application name: Duplicati.GUI.TrayIcon.exe, version: 2.0.2.1, time stamp: 0x5980511c
Faulting module name: KERNELBASE.dll, version: 10.0.15063.502, time stamp: 0x005405b5
Exception code: 0xe0434352
Fault offset: 0x0000000000069e08
Faulting process id: 0x554
Faulting application start time: 0x01d329bd2143e9ef
Faulting application path: C:\Program Files\Duplicati 2\Duplicati.GUI.TrayIcon.exe
Faulting module path: C:\WINDOWS\System32\KERNELBASE.dll
Report Id: ea48810e-cd3c-4ec9-b829-901cbe913eec
Faulting package full name: 
Faulting package-relative application ID:

**Comment by mentalinc** [comment]
duplicati-2.0.2.4_canary_2017-09-09-x64.msi has solved the issue for me
